cpp/arcticdb/codec/encode_common.hpp

```diff
@@ -55,11 +63,9 @@ struct BytesEncoder {
                 bytes_count,
                 1u,
                 data.block_and_offset(0).block_);
-            Encoder::encode(codec_opts, typed_block, *encoded_field, out_buffer, pos);
+            Encoder::encode(codec_opts, typed_block, encoded_field, out_buffer, pos);
         } else if constexpr (EncodingPolicyType::version == EncodingVersion::V2) {
-            // On Man's Mac build servers size_t and ssize_t are long rather than long long but the shape TDT
-            // expects int64 (long long).
-            const size_t row_count = 1;
+            const shape_t row_count = 1;  //TOD this seems weird and wrong. Why is it hardcoded to 1?
```

Comment:
If I remember correctly the BytesEncoder was used for things such as metadata and string pools this data is stored as an array type (DIM1) in the corresponding segment and the whole data is on one row.